hack/Makefile.debug

```diff
@@ -2,29 +2,137 @@
 
 export GOOS=linux
 
-REGISTRY ?= quay.io
-IMAGE ?= openshift/openshift-router
+REGISTRY ?= <some-registry.com>
+IMAGE ?= <some-user-namespace>/openshift-router
 TAG ?= latest
 IMAGEBUILDER ?= podman
 
-new-openshift-router-image:
+SERVICE_FILES := $(wildcard hack/remote-debug/*.service)
+SYSTEMD_DIR   := $(HOME)/.config/systemd/user
+
+DOCKERFILE_DEBUG ?= hack/Dockerfile.debug
+
+ifndef HAPROXY_RPMS
+  $(error HAPROXY_RPMS is not set. Please set the environment variable HAPROXY_RPMS. (e.g., HAPROXY_RPMS=https://github.com/frobware/haproxy-builds/raw/master/rhaos-4.17-rhel-9/haproxy28-2.8.10-1.rhaos4.17.el9.x86_64.rpm))
+endif
+
+# Uncomment the following for remote debugging.
+# DOCKERFILE_DEBUG = hack/remote-debug/Dockerfile
+
+ifeq ($(DOCKERFILE_DEBUG),hack/remote-debug/Dockerfile)
+  ifndef GITHUB_USER
+    $(error GITHUB_USER is not set. Please set the environment variable GITHUB_USER)
+  endif
+  # Remote debugging requires the dlv executable in the local
+  # directory which is copied into the container in
+  # hack/remote-debug/Dockerfile.
+  ifeq (,$(wildcard ./dlv))
+    $(error dlv executable not found in the current directory. Consider running ./hack/remote-debug/find-and-copy-dlv-from-goland-installation)
+  endif
+  # GITHUB_USER is a hard requirement. The docker build in
+  # hack/remote-debug/Dockerfile needs to fetch your public ssh keys
+  # from https://github.com/<GITHUB_USER>.keys.
+  export GITHUB_USER
+endif
+
+# The `--continue` flag for `dlv` (Delve debugger) starts the debuggee
+# immediately, without halting for a debugger to attach. This setup
+# ensures that the router is operational from the get-go, yet still
+# allows for remote debugging. Attach from your IDE at any time after
+# launch to begin debugging.
+DLV_CONTINUE ?= --continue
+
+build:
 	GO111MODULE=on CGO_ENABLED=0 GOFLAGS=-mod=vendor go build -o openshift-router -gcflags=all="-N -l" ./cmd/openshift-router
-	$(IMAGEBUILDER) build -t $(IMAGE):$(TAG) -f hack/Dockerfile.debug .
+	GO111MODULE=on CGO_ENABLED=0 GOFLAGS=-mod=vendor go build -o hack/remote-debug/remote-debug-helper -gcflags=all="-N -l" ./hack/remote-debug/remote-debug-helper.go
+
+build-image: build
+	$(IMAGEBUILDER) build --build-arg HAPROXY_RPMS="$(HAPROXY_RPMS)" --build-arg GITHUB_USER="$(GITHUB_USER)" -t $(IMAGE):$(TAG) -f $(DOCKERFILE_DEBUG) .
```

Comment:
I could dispense with the pulling keys from github and just create a temporary password-less key/pair, copying the public key into the container. It would be easier and one less thing to setup (in terms of the environoment).

Reply:
https://github.com/openshift/router/pull/609/commits/bc2d74554d59393bf0e76bb82ef93c28cf44b1a9 dynamically generates keys and no longer relies on or assumes that you use ssh keys for github.